x-pack/legacy/plugins/lens/public/datatable_visualization_plugin/visualization.tsx

```diff
@@ -115,7 +115,8 @@ export const datatableVisualization: Visualization<
   switchVisualizationType: (_, state) => state,
 
   initialize(frame, state) {
-    const layerId = Object.keys(frame.datasourceLayers)[0] || frame.addNewLayer();
+    const layerId = frame.addNewLayer();
```

Comment:
It looks like this is always initializing a new layer even when it's not necessary because the state is already set. This might cause an unnecessary layer when saving and loading a table vis

Reply:
It is unnecessary. Good call.